src/views/Home.jsx

```diff
@@ -1,7 +1,41 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList';
+import { useEffect, useState } from 'react';
+import { createList } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, setListPath, user }) {
+	const [shoppingListName, setShoppingListName] = useState('');
+	const navigate = useNavigate();
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		const { uid, email } = user;
+
+		const names = data.map((list) => {
+			return list.name;
+		});
+
+		if (names.includes(shoppingListName)) {
+			alert('name should be unique');
+
+			return;
+		}
+
+		createList(uid, email, shoppingListName);
```

Comment:
`createList` is a promise so we'll need some async handling and error handling here. A console log and an alert for the error seem fine to me as long as everyone agrees.